screeps-profiler.js

```diff
@@ -191,17 +219,31 @@ const Profiler = {
     { name: 'RoomPosition', val: RoomPosition },
     { name: 'Source', val: Source },
     { name: 'Flag', val: Flag },
+    { name: 'Market', val: Game.market },
   ],
 
-  record(functionName, time) {
+  record(myparent, functionName, time) {
     if (!Memory.profiler.map[functionName]) {
       Memory.profiler.map[functionName] = {
-        time: 0,
+        time: [],
```

Comment:
This isn't an array by design. The profiler is meant to run against any codebase for up to Infinity ticks.  I don't want to have an unbounded array for functions that are called many hundreds of thousands of times.